test/format/snap.c

```diff
@@ -659,7 +659,19 @@ snap_repeat(TINFO *tinfo, SNAP_OPS *snap)
 
         testutil_check(session->rollback_transaction(session, NULL));
     }
-    testutil_assert(max_retry < MAX_RETRY_ON_ROLLBACK);
+
+    /*
+     * If we have a long running checkpoint, it may block eviction for an excessive amount of time.
+     * This would cause the snapshot read to rollback even we retry many times. Give up and ignore
+     * this case.
+     */
+    if (max_retry >= MAX_RETRY_ON_ROLLBACK &&
+      strcmp(session->get_rollback_reason(session), WT_TXN_ROLLBACK_REASON_OLDEST_FOR_EVICTION) ==
```

Comment:
It's interesting to use the defined string, but definitely a layering violation. That constant is defined in `txn.h`, which is an internal header.
OTOH: It's nice to have this code be robust to changes in the string.
FTR, MongoDB has it's own definition of the string here:
https://github.com/mongodb/mongo/blob/master/src/mongo/db/storage/wiredtiger/wiredtiger_util.cpp#L54

Reply:
ideally we should extract the definition to a separate common header that is shared between wt core library and the tests. For this specific case, I think it makes sense to define in the api header as the user actually needs it. For now, I'll just redefine it in this file.